### PR TITLE
feat(oxc_linter): implement `eslint(for-direction)`

### PR DESCRIPTION
**Pull request by YangchenYe323**
Implement ESLint's [for-direction](https://eslint.org/docs/latest/rules/for-direction) rule.

### CODE REVIEW

File: crates/oxc_linter/src/rules/for_direction.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        _ => return UNKNOWN,
+    };
+
+    if !positive {
+        direction = -direction;
+    }
+    direction
+}
+
+impl Rule for ForDirection {
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/for_direction.rs:103]: Let's move this block to the top, just under `declare_oxc_lint` so it's easier for people to read and navigate.

File: crates/oxc_linter/src/snapshots/for_direction.snap [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+source: crates/oxc_linter/src/tester.rs
+expression: for_direction
+---
+
+  ⚠ eslint(for-direction): Disallow for loop in the wrong direction
+   ╭─[for_direction.tsx:1:1]
+ 1 │ for (var a = 0; a < 10; a--) {}
+   · ───────────────────────────────
+   ╰────
+  help: Use while loop for intended infinite loop
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/for_direction.snap:11]:
```
  ⚠ eslint(for-direction): The update clause in this loop moves the variable in the wrong direction
   ╭─[for_direction.tsx:1:16]
 1 │ for(var i = 0; i < 10; i--){}
   ·                ───┬──  ─┬─
   ·                   │     ╰── with this update
   ·                   ╰── This test moves in the wrong direction
   ╰────
```

I think I need to write up a guide on what to point at, but basically we want to minimize the span to point at for context.

File: crates/oxc_linter/src/rules/for_direction.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    let fail = vec![
+        ("for (var a = 0; a < 10; a--) {}", None),
+        ("for (var a = 0; a < 10; a += -1) {}", None),
+        ("for (var a = 0; a < 10; a -= 1) {}", None),
+        ("for (; a >= 0; a++) {}", None),
+        ("for (; a >= 0; a -= -1) {}", None),
+        ("for (; a >= 0; a += 1) {}", None),
+        ("for (; a < 10; a += -foo()) {}", None),
+    ];
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/for_direction.rs:153]: I think we should copy eslint's tests, so we don't get lost.

File: crates/oxc_linter/src/rules/for_direction.rs (hunk 1 above)
Comment (YangchenYe323) [reply crates/oxc_linter/src/rules/for_direction.rs:103]: Done.

File: crates/oxc_linter/src/snapshots/for_direction.snap (hunk 2 above)
Comment (YangchenYe323) [reply crates/oxc_linter/src/snapshots/for_direction.snap:11]: Fixed.

File: crates/oxc_linter/src/rules/for_direction.rs (hunk 3 above)
Comment (YangchenYe323) [reply crates/oxc_linter/src/rules/for_direction.rs:197]: Done.

**Comment by Boshen** [comment]
Awesome! Thank you!

**Pull request merged**